app/[blog]/new/page.js

```diff
@@ -0,0 +1,91 @@
+"use client";
+
+import { useState } from "react";
+import { useSearchParams } from "next/navigation";
+import Markdown from "marked-react";
+
+import "@uiw/react-markdown-preview/markdown.css";
+import * as commands from "@uiw/react-md-editor/lib/commands";
+
+import "@uiw/react-md-editor/markdown-editor.css";
+import dynamic from "next/dynamic";
+import { useRouter } from "next/navigation";
+
+import styles from "./page.module.css";
+
+const MDEditor = dynamic(() => import("@uiw/react-md-editor"), { ssr: false });
+
+const editorOptions = {
+  preview: "edit",
+  commands: [
+    "bold",
+    "italic",
+    "strikethrough",
+    "hr",
+    "divider",
+    "link",
+    "quote",
+    "code",
+    "codeBlock",
+    "image",
+    "divider",
+    "unorderedListCommand",
+    "orderedListCommand",
+    "checkedListCommand",
+  ].map((k) => commands[k]),
+  extraCommands: [],
+};
+
+// mock data/api
+let blogPost = {};
+const useCurrentUser = () => ({ name: useSearchParams().get("user") }); // `?user=TestUser` in URL to test author view
+const submitBlogPost = (name, title, slug, text) =>
+  (blogPost = { postId: text.length, authorUsername: name, title: title, slug: slug, text: text });
+// end mock
+
+const NewBlogPost = () => {
+  const router = useRouter();
+  const { name: authorUsername } = useCurrentUser();
+  const [text, setText] = useState("");
+  const [title, setTitle] = useState("");
+
+  const getSlugFromTitle = (title) => title.trim().replaceAll(" ", "-");
+
+  const handleSubmit = () => {
+    const slug = getSlugFromTitle(title);
```

Comment:
might not be a good idea for client to supply the slug as it needs to unique in db

Reply:
yeah true, this could still be part of the request and the backend could add an identifier in case of any collisions (which should be rare if this is by blog but still). and then we can use that in the response here and redirect